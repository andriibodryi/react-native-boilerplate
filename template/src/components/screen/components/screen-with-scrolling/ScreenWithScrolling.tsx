import * as React from 'react';
import { View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import { useScreenWithScrollingStyles } from '@src/components/screen/components/screen-with-scrolling/screen-with-scrolling.presets';
import { useStyleConfig } from '@src/styles/useStyleConfig';

import { ScreenTypes } from '../../screen.types';
import { offsets } from '../../screen.presets';

const ScreenWithScrolling: React.FC<ScreenTypes> = ({
  backgroundColor,
  children,
  keyboardOffset,
  keyboardShouldPersistTaps,
  bounces,
  style = {},
}) => {
  const { colors } = useStyleConfig();
  const styles = useScreenWithScrollingStyles();

  const backgroundStyle = {
    backgroundColor: backgroundColor || colors.white,
  };

  return (
    <View style={[styles.outer, backgroundStyle]}>
      <KeyboardAwareScrollView
        style={[styles.outer, backgroundStyle]}
        contentContainerStyle={[styles.inner, style]}
        keyboardShouldPersistTaps={keyboardShouldPersistTaps || 'handled'}
        extraHeight={offsets[keyboardOffset || 'none']}
        bounces={bounces}
      >
        {children}
      </KeyboardAwareScrollView>
    </View>
  );
};

export default ScreenWithScrolling;
