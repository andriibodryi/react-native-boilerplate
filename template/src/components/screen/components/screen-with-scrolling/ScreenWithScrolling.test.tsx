import React from 'react';
import renderer from 'react-test-renderer';

import ScreenWithScrolling from './ScreenWithScrolling';

describe('ScreenWithScrolling', () => {
  it('should render scrollable screen', () => {
    const tree = renderer
      .create(<ScreenWithScrolling preset="scroll" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
