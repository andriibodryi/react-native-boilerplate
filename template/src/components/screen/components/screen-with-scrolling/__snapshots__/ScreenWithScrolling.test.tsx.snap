// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenWithScrolling should render scrollable screen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  Test child
</View>
`;
